internal/protoveneer/support/support.go

```diff
@@ -119,3 +122,21 @@ func TimeFromProto(ts *timestamppb.Timestamp) time.Time {
 	}
 	return ts.AsTime()
 }
+
+// APIErrorToProto attempts to convert an APIError to a proto Status.
+func APIErrorToProto(ae *apierror.APIError) *spb.Status {
+	if ae == nil {
+		return nil
+	}
+	return ae.GRPCStatus().Proto()
+}
+
+func APIErrorFromProto(s *spb.Status) *apierror.APIError {
```

Comment:
@noahdietz is there a more direct path here than from proto status -> grpc status -> error -> APIError?

Reply:
There isn't, but we can totally add one!

The use case for `APIError` first targeted wrapping API error responses which for GAPIC are always either a `grpc.Status` (wrapping the `proto.Status`) or a `googleapi.Error` (from the Discovery-based generated HTTP/JSON clients).

If you want to add it yourself, feel free, otherwise please [file an issue on gax-go](https://github.com/googleapis/gax-go/issues/new)